Plugins/NativeCamera/Source/Apple/NativeCameraImpl.mm

```diff
@@ -39,13 +40,27 @@ - (id)init:(std::shared_ptr<Babylon::Plugins::Camera::Impl::ImplData>)implData;
                 CVMetalTextureCacheFlush(textureCache, 0);
                 CFRelease(textureCache);
             }
+            for (auto textureRef : texture)
+            {
+                if (textureRef)
+                {
+                    CFRelease(textureRef);
+                }
+            }
         }
-        
+
         CameraTextureDelegate* cameraTextureDelegate{};
         AVCaptureSession* avCaptureSession{};
         CVMetalTextureCacheRef textureCache{};
+        /*
+         AVfoundation might update a texture while it's still being processed by the rendering. If it happens, slowdowns or no update at all will occur.
+         To fix this issue, a texture array is used with enough slots. 1 texture is used for rendering while another is updated.
```

Comment:
This explanation sounds like there should be 2 textures. Why 4?

Reply:
With 2 textures, I still get shuttering sometimes. 3 is ok but I always favor power of 2. Which brings a question. For Apple and other platform, when user is updating a texture outside of BN, can it be a problem? For example, when a user updates a texture faster than rendering on another thread? Or can it happen in the middle of bgfx rendering?